### PR TITLE
Home Page Styling, Layout, Paritals

### PR DESCRIPTION
**Pull request by OHaiDanny**
homepageNov4

In This PR I've updated the homepage to look like what is shown in the mockups

- Added some logic to use a unique header for the homepage only.
- Added partials and data for the encounter section of the homepage
- Added content for the role slider and added some JS to add a unique classname for styling
- lots of tailwind classes to style the page

Let me know if theres any questions or comments :)

### CODE REVIEW

File: assets/js/role_slider.js
Comment (nonowazu) [inline assets/js/role_slider.js:12]: Out of curiosity, why the JS? From what I remember, the role slider needed no js for either theming or operation.

File: data/encounter/savage.json
Comment (nonowazu) [inline data/encounter/savage.json:1]: Could you add a way to generate encounter data in `exampleSite/static/admin/config.yml`? You can find docs for this [here](https://www.netlifycms.org/docs/intro/)

File: data/encounter/savage.json
Comment (nonowazu) [inline data/encounter/savage.json:7]: I *think* these might not be necessary, since I think you can do the following in hugo, but we should probably brainstorm on using `range` with hugo on data to make this auto-iterate. I'll take a look a bit later, but if this ends up being required, then it's required.

File: data/frontpage/slider.json
Comment (nonowazu) [inline data/frontpage/slider.json:27]: Goodbye, sweet joke.

File: layouts/_default/baseof.html
Comment (nonowazu) [inline layouts/_default/baseof.html:11]: You can do an [if..else](https://gohugo.io/templates/introduction/#example-4-if--else) in hugo for this whole part

File: layouts/index.html
Comment (nonowazu) [inline layouts/index.html:4]: While we're going through this PR, I might ask you to hold off on some of these partials – get that the home page looks empty without some of these, but at the same time we've also got to focus on getting the theme good enough to move over to the primary domain/repo.

File: layouts/partials/homepage/encounters/card.html
Comment (nonowazu) [inline layouts/partials/homepage/encounters/card.html:8]: I feel like some of this conflicts with the other active PR right now – you might want to discuss how to handle this with them. They technically submitted their PR first so I'm going to give them first dibs at merging, which might mess up your flow with card partials.

File: layouts/partials/homepage/role_slider.html
Comment (nonowazu) [inline layouts/partials/homepage/role_slider.html:1]: I might ask that you go into the css file and just `@apply` the changes to the tabbed class instead of adding them here.

File: layouts/partials/homepage/role_slider/content.html
Comment (nonowazu) [inline layouts/partials/homepage/role_slider/content.html:1]: ditto on `@apply`ing this in the `.tab-content` selector in the main css.

File: static/theme-assets/icons/role/role_melee.svg
Comment (nonowazu) [inline static/theme-assets/icons/role/role_melee.svg:4]: The second line is drawing incorrectly here.

File: tailwind.config.js
Comment (nonowazu) [inline tailwind.config.js:21]: Ditto on coordinating with Roxas re: changes here – I'd rather have both of you be happy than me be miserable sorting it out :-)

**Review by nonowazu — CHANGES_REQUESTED** [review]

File: assets/js/role_slider.js
Comment (OHaiDanny) [reply assets/js/role_slider.js:12]: Ah yes, so I noticed that in the mockups the "selected" tab has the same background image as the body of the role slider. I think this is to visually show to the user which tab they have selected when they click around them. I was trying to figure out if there was a way using hugo to add this "selected" class name.

My knowledge on hugo is very basic so I thought maybe it would be easier to just some JS to add that class instead.

File: data/encounter/savage.json
Comment (OHaiDanny) [reply data/encounter/savage.json:7]: Sounds good :)

File: data/frontpage/slider.json
Comment (OHaiDanny) [reply data/frontpage/slider.json:27]: aww 😢

File: tailwind.config.js
Comment (Roxas92) [reply tailwind.config.js:21]: @OHaiDanny once my PR is merged I can take care of these colors in your PR as well. We most likely used the same color codes but with different naming. I'm more a fan of not having words like "color" inside of color-variables since writing them unser color imply that they serve as color suffixes.

File: layouts/partials/homepage/encounters/card.html
Comment (Roxas92) [reply layouts/partials/homepage/encounters/card.html:8]: @OHaiDanny same as in my other comment: If you want to I can check this partial once my PR is merged.

File: assets/js/role_slider.js
Comment (nonowazu) [reply assets/js/role_slider.js:12]: Oh, I see what I missed: if you'll look [here](https://github.com/The-Balance-FFXIV/glam/blob/9a716f4ae94b847536235643b977c2c6a60da74d/assets/css/main.css#L95), I commented out the styling for tabs which are "checked". With that, you should be able to remove all the js and the tabs become a css-only affair (which I'll fully admit I was needlessly proud of).

Try uncommenting out that section and moving your styles there?

File: layouts/partials/homepage/role_slider.html
Comment (OHaiDanny) [reply layouts/partials/homepage/role_slider.html]: addressed!

File: layouts/_default/baseof.html
Comment (OHaiDanny) [reply layouts/_default/baseof.html:11]: addressed in upcoming code push

File: layouts/index.html
Comment (OHaiDanny) [reply layouts/index.html:4]: Sounds good. I believe I'll comment out the "Latest Guides" and the "Community Spotlight" sections as that is what the consensus is from the dev channel :)

File: layouts/partials/homepage/role_slider/content.html
Comment (OHaiDanny) [reply layouts/partials/homepage/role_slider/content.html]: updated!

File: tailwind.config.js
Comment (OHaiDanny) [reply tailwind.config.js]: I've updated the colors to use what @Roxas92 merged in :)

File: assets/js/role_slider.js
Comment (OHaiDanny) [reply assets/js/role_slider.js:12]: Will have an update in the next PR!

File: data/encounter/savage.json
Comment (OHaiDanny) [reply data/encounter/savage.json]: @nonowazu I made a refactor of the data structure here to accomodate the added code in the `admin/config.yml`. Please let me know if I missed anything (first time doing the config.yml stuff)

File: assets/js/role_slider.js
Comment (OHaiDanny) [reply assets/js/role_slider.js]: This has been updated in the latest PR!

File: data/encounter/savage.json
Comment (OHaiDanny) [reply data/encounter/savage.json:1]: Updated in commit https://github.com/The-Balance-FFXIV/glam/pull/63/commits/6f3ed701a95c5a1704ecec6552bb2c2833a01974

File: exampleSite/static/admin/config.yml
Comment (nonowazu) [inline exampleSite/static/admin/config.yml:68]: I'm seeing this when I use hugo to render and then go to http://localhost:1313/admin/#/:

Screen Shot 2021-11-05 at 6 17 37 PM

I believe it's because you're doing json-like `{}` after the `fields:` view; maybe something more like this?

```yaml
fields:
  - {label: "Identifier", name: "id", widget: "string"}
  - {label: "Name of Encounter", name: "name", widget: "string"}
  - {label: "Image Link", name: "image_link", widget: "string"}
  - {label: "Encounter Link", name: "encounter_link", widget: "string"}
  ```
I might also switch the string widget out for `select` widget (docs [here](https://www.netlifycms.org/docs/widgets/#select)) – this way, we can narrow the category of file(s).

tl;dr - yaml is weird and we should probably make a PR checker that'll yamllint this file just like we have on the production repo.

File: assets/js/role_slider.js
Comment (nonowazu) [inline assets/js/role_slider.js:12]: One last nit on this, I swear!

If the css-only setup worked, would you mind removing this file (and from the associated head partial)?

**Review by nonowazu — CHANGES_REQUESTED** [review]
Just 2 small changes and I think it's good to go!

File: assets/js/role_slider.js
Comment (OHaiDanny) [reply assets/js/role_slider.js:12]: Oh shoot I forgot to remove this! My bad!

File: exampleSite/static/admin/config.yml
Comment (OHaiDanny) [reply exampleSite/static/admin/config.yml:68]: ahhh my bad. You're correct; the `fields:` shouldn't start with `{` oops! I'll try to test this on the admin page before I push the change.

Comment (OHaiDanny) [reply exampleSite/static/admin/config.yml]: @nonowazu fixed in commit https://github.com/The-Balance-FFXIV/glam/pull/63/commits/f9fbce0568dc5a840fde1d8445d24742fc61b1d3

File: assets/js/role_slider.js
Comment (OHaiDanny) [reply assets/js/role_slider.js]: @nonowazu fixed in commit https://github.com/The-Balance-FFXIV/glam/pull/63/commits/f9fbce0568dc5a840fde1d8445d24742fc61b1d3

**Review by nonowazu — APPROVED** [review]

**Pull request merged**
